src/Item/Item.service.ts

```diff
@@ -120,38 +121,66 @@ export class ItemService {
   }
 
   public async getCollectionItems(
-    collectionId: string
-  ): Promise<{ collection: CollectionAttributes; items: FullItem[] }> {
-    const [dbCollection, dbItems] = await Promise.all([
-      this.collectionService.getDBCollection(collectionId),
-      Item.find<ItemAttributes>({ collection_id: collectionId }),
-    ])
-
-    return isTPCollection(dbCollection)
-      ? this.getTPCollectionItems(dbCollection, dbItems)
-      : this.getDCLCollectionItems(dbCollection, dbItems)
+    collectionId: string,
+    filters: {
+      status?: CurationStatus
+      limit?: number
+      offset?: number
+    }
+  ): Promise<{
+    collection: CollectionAttributes
+    items: FullItem[]
+    totalItems: number
+  }> {
+    const { status, limit, offset } = filters
+    const dbCollection = await this.collectionService.getDBCollection(
+      collectionId
+    )
+    const isTP = isTPCollection(dbCollection)
+    const dbItems =
+      status && isTP
+        ? await Item.findByCollectionIdAndPendingToApprove(
+            collectionId,
+            limit,
+            offset
+          )
+        : await Item.findByCollectionIds([collectionId], limit, offset)
+
+    const { collection, items } = isTPCollection(dbCollection)
+      ? await this.getTPCollectionItems(dbCollection, dbItems)
+      : await this.getDCLCollectionItems(dbCollection, dbItems)
+
+    return {
+      collection,
+      items,
+      totalItems: Number(dbItems[0]?.total_count ?? 0),
+    }
   }
 
-  public async getTPItemsByManager(manager: string): Promise<ItemAttributes[]> {
-    const thirdParties = await thirdPartyAPI.fetchThirdPartiesByManager(manager)
+  public async getStandardAndTPItems(
```

Comment:
I'm not sure about this method. I understand the need for it, but it's doing a bit too much to me. Or at least, it's not explicit enough about it
What I mean by that is that the method is only receiving an address but it's called `getStandardAndTPItems` so it's not clear what's being used for. Same goes for `findStandardAndTPItems`

I was thinking what to do about it and the best I can come up with (without spliting the methods) is something like: `findAllForAddress`. What do you think?

@LautaroPetaccio

Reply:
Although possible, it's pretty hard to split the methods in two (one for the TP items and the other one for the Standard ones) and keep the pagination working. I think this method is the fastest way to implement pagination without worrying about any edge case that merging the two methods might have. We can rename the method to `findAllForAddress` and we can also add JSDoc for this method or others.
Saying all of this, `getTPItemsByManager` should not be deleted, as we'll surely have filters to retrieve only the TP or standard items.